lib/s3-html/log.html

```diff
@@ -406,38 +406,45 @@
     return null;
 }
 
+class StopError extends Error { };
```

Comment:
I think if we get any hard error (basically the branch where we return `null` in `fetch_content()` then we should throw this exception from that point (possibly with a comment to the effect).

That would let us remove the conditions on calling `fetch_content()`.  Probably we'd then rewrite the `while()` loop as some infinite iteration or so.